code2seq/data/comment_path_context_dataset.py

```diff
@@ -0,0 +1,19 @@
+from typing import Dict, List, Optional
+
+from transformers import RobertaTokenizerFast
+
+from code2seq.data.path_context_dataset import PathContextDataset
+
+tokenizer = RobertaTokenizerFast.from_pretrained("microsoft/codebert-base")
+
+
+class CommentPathContextDataset(PathContextDataset):
+    @staticmethod
+    def tokenize_label(raw_label: str, vocab: Dict[str, int], max_parts: Optional[int]) -> List[int]:
+        label_with_spaces = " ".join(raw_label.split(PathContextDataset._separator))
+        label_tokens = tokenizer.tokenize(label_with_spaces)
+        if max_parts is None:
+            max_parts = len(label_tokens)
+        label_tokens = [tokenizer.bos_token] + label_tokens[: max_parts - 2] + [tokenizer.eos_token]
+        label_tokens += [tokenizer.pad_token] * (max_parts - len(label_tokens))
```

Comment:
The whole function body can be replaced with a single call to tokenizer:
```python
tokenized_snippet = tokenizer(
    raw_label.split(PathContextDataset._separator),
    add_special_tokens=True, 
    padding="max_length" if max_parts else "do_not_pad", 
    max_length=max_parts,
    is_split_into_words=True,
)
```

In order for `is_split_into_words=True` to work, add `add_prefix_space=True` to tokenizer creation.

Reply:
The output of tokenizer will be a dict with `input_ids` for tokens and `attention_mask` to point out which tokens come from padding. If you don't need the mask, pass `return_attention_mask=False`.